### PR TITLE
[Consensus] Add counters for consensus messages by type

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

This will be useful to understand the breakdown of different messages and see if there is room for optimization to reduce the volume of consensus messages. 

### Test Plan

Existing UT.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3900)

### CODE REVIEW

**Review by zekun000 — COMMENTED** [review]
let's add a counter for messages we sent as well?

**Comment by sitalkedia** [comment]
> let's add a counter for messages we sent as well?

Let me see if there is a single place to add that - I can do that as a followup PR though. Thoughts?

**Comment by igor-aptos** [comment]
@sitalkedia check the consensus::send::all failpoints, it should all be in one file

**Comment by sitalkedia** [comment]
> consensus::send::all

Yes, added counters for sent traffic as well in the latest iteration. PTAL

**Review by igor-aptos — COMMENTED** [review]
(cannot comment inline for some reason:

you missed the BlockRetrievalRequest in request_block() in network.rs

File: consensus/src/network_interface.rs [hunk 1]
```diff
@@ -64,6 +65,23 @@ pub enum ConsensusMsg {
     CommitDecisionMsg(Box<CommitDecision>),
 }
 
+impl Display for ConsensusMsg {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```
Comment (zekun000) [inline consensus/src/network_interface.rs:69]: we probably don't want this in Display, this will make all the debug print omit the actual message content

Comment (sitalkedia) [reply consensus/src/network_interface.rs]: We are not using Display anywhere for this for now but sure changed it.

**Review by zekun000 — APPROVED** [review]

**Review by igor-aptos — APPROVED** [review]

**Pull request merged**